res/controllers/Denon-MC7000-scripts.js

```diff
@@ -573,34 +610,154 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
-                        for (j = 1; j <=8; j++) {
-                            engine.setValue("[Sampler" + j + "]", "cue_gotoandstop", 1);
+                        // stop playing all other samplers on this deck
+                        for (j = 1; j <= 8; j++) {
+                            samplerOffsetJ = deckOffset * 8 + j;
+                            if (engine.getValue("[Sampler" + samplerOffsetJ + "]", "play") === 1) {  // if sampler is playing then stop it
+                                engine.setValue("[Sampler" + samplerOffsetJ + "]", "cue_gotoandstop", 1);
+                                midi.sendShortMsg(0x94 + deckOffset, 0x14 + j - 1, MC7000.padColor.samplerloaded);
+                            }
                         }
                     }
                     // ... before the actual sampler to play gets started
-                    engine.setValue("[Sampler" + i + "]", "cue_gotoandplay", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "pregain", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "cue_gotoandplay", 1);
+                    midi.sendShortMsg(0x94 + deckOffset, 0x14 + i - 1, MC7000.padColor.samplerplay);
+                }
+            } else if (control === 0x1C + i - 1 && value >= 0x01) { //shifted button deactivates playing sampler or ejects sampler
+                if (MC7000.SamplerQty === 16) {
+                    if ((deckOffset) >= 2) {
+                        deckOffset = deckOffset + 2;
+                    }
                 }
-            } else if (control === 0x1C + i - 1 && value >= 0x01) {
-                if (engine.getValue("[Sampler" + i + "]", "play") === 1) {
-                    engine.setValue("[Sampler" + i + "]", "cue_gotoandstop", 1);
+                if (engine.getValue("[Sampler" + samplerOffset + "]", "play") === 1) {
+                    engine.setValue("[Sampler" + samplerOffset + "]", "cue_gotoandstop", 1);
                     midi.sendShortMsg(0x94 + deckOffset, 0x1C + i - 1, MC7000.padColor.samplerloaded);
                 } else {
-                    engine.setValue("[Sampler" + i + "]", "eject", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "eject", 1);
                     midi.sendShortMsg(0x94 + deckOffset, 0x1C + i - 1, MC7000.padColor.sampleroff);
-                    engine.setValue("[Sampler" + i + "]", "eject", 0);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "eject", 0);
+                }
+            }
+        }
+    } else if (MC7000.PADModeVelSamp[deckOffset]) {
+        samplerOffset = 0;
+        samplerOffsetJ = 0;
+        for (i = 1; i <= 8; i++) {
+            if (MC7000.SamplerQty === 16) {
+                if (deckOffset >= 2) {
+                    deckOffset = deckOffset - 2;
+                }
+            }
+            samplerOffset = deckOffset * 8 + i;
+            if (control === 0x14 + i - 1 && value >= 0x01) {
```

Comment:
This if branch needs a comment, that explains, when it's entered.

Reply:
done